finance-service-api/src/test/kotlin/com/ajaxproject/financeservice/NatsControllerTest.kt

```diff
@@ -0,0 +1,188 @@
+import com.ajaxproject.financeservice.enums.Finance.EXPENSE
+import com.ajaxproject.financeservice.enums.Finance.INCOME
+import com.ajaxproject.financeservice.models.MongoFinance
+import com.ajaxproject.internalapi.finance.input.reqreply.CreateFinanceRequest
+import com.ajaxproject.internalapi.finance.input.reqreply.CreateFinanceResponse
+import com.ajaxproject.internalapi.finance.input.reqreply.DeleteFinanceByIdRequest
+import com.ajaxproject.internalapi.finance.input.reqreply.DeleteFinanceByIdResponse
+import com.ajaxproject.internalapi.finance.input.reqreply.GetAllFinancesByIdRequest
+import com.ajaxproject.internalapi.finance.input.reqreply.GetAllFinancesByIdResponse
+import com.ajaxproject.internalapi.finance.input.reqreply.GetCurrentBalanceRequest
+import com.ajaxproject.internalapi.finance.input.reqreply.GetCurrentBalanceResponse
+import com.google.protobuf.GeneratedMessageV3
+import com.google.protobuf.Parser
+import io.nats.client.Connection
+import org.assertj.core.api.Assertions.assertThat
+import org.bson.types.ObjectId
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.query.Criteria
+import org.springframework.data.mongodb.core.query.Query
+import java.time.Duration
+import java.util.*
+
+
+@SpringBootTest
+class NatsControllerTest {
+
+    @Autowired
+    private lateinit var natsConnection: Connection
+
+    @Autowired
+    private lateinit var mongoTemplate: MongoTemplate
+
+    @Autowired
+    private lateinit var financeRepository: FinanceRepositoryImpl
+
+    @AfterEach
+    fun clean() {
+        val query = Query.query(Criteria.where("userId").`is`(4757839801))
+        mongoTemplate.remove(query, MongoFinance::class.java)
+    }
+
+    private val testMongoIncomeFinance: MongoFinance =
+        MongoFinance(
+            id = ObjectId(),
+            userId = 4757839801,
+            financeType = INCOME,
+            amount = 100.0,
+            description = "Test Description",
+            date = Date()
+        )
+
+
+    @Test
+    fun `should return expected finance when get all finance by request`() {
+        //GIVEN
+        financeRepository.save(testMongoIncomeFinance)
+        val userIdSave = testMongoIncomeFinance.userId
+        val financeTypeSave = testMongoIncomeFinance.financeType.toProtoEnumFinance()
+
+        val request = GetAllFinancesByIdRequest.newBuilder()
+            .setUserId(userIdSave)
+            .setFinanceType(financeTypeSave)
+            .build()
+
+        val expectedResponse = GetAllFinancesByIdResponse.newBuilder().apply {
+            successBuilder.addAllFinance(listOf(testMongoIncomeFinance.toProtoFinance()))
+        }.build()
+
+        //WHEN
+        val currentsResponse = doRequest(
```

Comment:
maybe it is better to call it `actualResponse`?

Reply:
Agree